optihood/plot_functions.py

```diff
@@ -1,1027 +1,1031 @@
+        "(('electricityInBus', 'HP'), 'flow')": "HP",
+        "(('electricityInBus', 'GWHP'), 'flow')": "GWHP",
+        "(('CHP', 'electricityProdBus'), 'flow')": "CHP_elec",
+        "(('electricalStorage', 'electricityBus'), 'flow')": "Battery_out",
+        "(('electricitySource', 'electricityBus'), 'flow')": "Elect_produced (not stored)",
+        "(('electricityResource', 'gridBus'), 'flow')": "Grid_purchase",
+        "(('dhwStorage', 'domesticHotWaterBus'), 'flow')": "Storage_dhw_out",
+        "(('dhwStorageBus', 'dhwStorage'), 'flow')": "Storage_dhw_in",
+        "(('domesticHotWaterBus', 'domesticHotWaterDemand'), 'flow')": "Demand_dhw",
+        "(('HP', 'dhwStorageBus'), 'flow')": "HP_dhw",
+        "(('GWHP', 'dhwStorageBus'), 'flow')": "GWHP_dhw",
+        "(('CHP', 'dhwStorageBus'), 'flow')": "CHP_dhw",
+        "(('shSource', 'spaceHeatingBus'), 'flow')": "SH_produced (not stored)",
+        "(('shStorage', 'spaceHeatingBus'), 'flow')": "Storage_sh_out",
+        "storage_content": "Storage_content",
+        "(('shSourceBus', 'shStorage'), 'flow')": "Storage_sh_in",
+        "(('spaceHeatingBus', 'spaceHeating'), 'flow')": "SH_direct_to_load",
+        "(('shDemandBus', 'spaceHeatingDemand'), 'flow')": "Demand_sh",
+        "(('CHP', 'shSourceBus'), 'flow')": "CHP_sh",
+        "(('GasBoiler', 'shSourceBus'), 'flow')": "Gas_sh",
+        "(('GasBoiler', 'dhwStorageBus'), 'flow')": "Gas_dhw",
+        "(('HP', 'shSourceBus'), 'flow')": "HP_sh",
+        "(('GWHP', 'shSourceBus'), 'flow')": "GWHP_sh",
+        "(('electricityBus', 'producedElectricity'), 'flow')": "Self_consumption",
+        "(('gridBus', 'gridElectricity'), 'flow')": "Electricity_grid",
+        "(('pv', 'electricityProdBus'), 'flow')": "PV_elec",
+        "(('solarCollector', 'dhwStorageBus'), 'flow')": "SolarCollector",
+        "(('electricityInBus', 'solarCollector'), 'flow')": "SolarCollector",
+        "(('gridElectricity', 'electricityInBus'), 'flow')": "Electricity_grid",
+        "(('producedElectricity', 'electricityInBus'), 'flow')": "Self_consumption",
+        "(('electricityProdBus', 'electricitySource'), 'flow')": "Battery_bypass"
+    }
+    #if optMode == "group":
+    newLegends["(('electricityBus', 'electricityLink'), 'flow')"] = "electricityLink_out"
+    newLegends["(('electricityLink', 'electricityInBus'), 'flow')"] = "electricityLink_in"
+    newLegends["(('spaceHeatingBus', 'shLink'), 'flow')"] = "shLink_in"
+    newLegends["(('shLink', 'shDemandBus'), 'flow')"] = "shLink_out"
+
+    resultFileBasePath = "..\data\Results"
+    resultFileName = f"results{numberOfBuildings}_{plotOptim}_{optMode}.xlsx"  # add the name of the excel file here for which the plots are to be made
+    if not os.path.exists(resultFileBasePath):
+        os.makedirs(resultFileBasePath)
+    resultFilePath = os.path.join(resultFileBasePath, resultFileName)
+
+    createPlot(resultFilePath, plotLevel, plotType, flowType, plotAnnualHorizontalBar, newLegends)
+

```

Comment:
Result filename and path should not be hardcoded. This should be corrected before the final merge (please look at the latest version of the code).